### ISSUE TITLE
Show each `User Interface Language` in its native language

### ISSUE THREAD

**Issue by ovari** [moneymanagerex/moneymanagerex#5410]
MMEX version: 1.6.2 Beta

Operating System: Linux Mint 21 Cinnamon

# Description of the bug
Each `User Interface Language` is always shown in English. Each `User Interface Language` should show in its native language as this would be a better User Experience.

Is the bug reproducible? Always 

Expected result:
Show each language in it's native language.

Actual result:
All languages are shown in English.

# Additional information
1. The order of the list the languages should be by their language code, eg `hu-HU`.
2. `Hungarian (Hungary)` should be shown as `magyar`.

![User Interface Language](https://user-images.githubusercontent.com/17465872/206665156-20d7fab1-cee2-4e42-9d86-d267e7a56a74.png)

What do you think?

Thank you

**Comment by vomikan** [comment]
I am personally against it. Let's say that in portable mode there will not be the right font to display national symbols, then this menu item will be mostly unreadable.

**Comment by ovari** [comment]
> Let's say that in portable mode there will not be the right font to display national symbols, then this menu item will be mostly unreadable.

Is your comment another reason to show native languages? If the correct font is unavailable to display native characters, would that indicate that the operating system does not fully support the script and users should not change to that language as the complete GUI may be unreadable?

In the footer of the [Thunderbird](https://www.thunderbird.net/) webpage, the `Page language` field shows in native languages.
![Thunderbird website Page language field](https://user-images.githubusercontent.com/17465872/206793239-01ff10cc-40df-4ea3-a6c3-27fe95bfbb91.png)

Perhaps each language in the menu could have the language code at the beginning (https://www.venea.net/web/culture_code)? Then if the script is not installed on the user's computer it will show strange characters and can help the user to install the required fonts before changing the `User Interface Language`. For example:
* `ar-SA العربية (المملكة العربية السعودية)`
* `de-DE Deutsch (Deutschland)`
* `fr-FR Français (France)`
* `hu-HU magyar (Magyarország)`
* `ru-RU Русский (Россия)`

What do you think?

Thank you

**Comment by ovari** [comment]
For more implementation ideas and screenshots from other programs, please see https://github.com/pdfarranger/pdfarranger/issues/706.

**Comment by ovari** [comment]
Can you please advise where the strings are located? This can help in creating a merge request with the language in the native language. Thank you

**Comment by n-stein** [comment]
> Can you please advise where the strings are located?

These strings are not stored anywhere in MMEX's codebase, they are provided directly by wxWidgets using the function wxLocale::FindLanguageInfo() to get the language descriptions for each locale code. The language names are not tagged for translation in the wxWidgets source, so the wxLocale object contains only the language names in English. 

We need to add all of the language name strings manually to something like constants.cpp and tag them for translation, then grab the translations of the descriptions returned by wxLocale::FindLanguageInfo() when building the app start dialog.

**Comment by ovari** [comment]
Thanks @n-stein for your comment. Is it possible for the code to be done and we can provide the strings in the constants.cpp file? Thank you

**Comment by n-stein** [comment]
> Is it possible for the code to be done and we can provide the strings in the constants.cpp file?

The downside of this approach is you would need to translate all language descriptions in all 34 Crowdin languages. It would be better to create a static po file with just the translated language list and add it as a second source for the translations catalog. This way you only need the translations once. Let me work on it.

**Comment by n-stein** [comment]
Done. Once #6171 is merged, you can just add the translations to the po/common.po file. I've included the translation you provided above in the file already as a reference sample, but feel free to change it as needed:

```
msgid "Hungarian (Hungary)"
msgstr "hu-HU magyar (Magyarország)"
```

The common.po file is appended to all of the normal *.po files prior to compilation so it gets included in all translation catalogs automatically.

**Comment by n-stein** [comment]
New PR is #6172.

**Comment by ovari** [comment]
Followup PR is https://github.com/moneymanagerex/moneymanagerex/pull/6173.

**Comment by ovari** [comment]
1. Works if the language is set to `hu-HU magyar (Magyarország)`
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/6448af59-42cb-4a73-b850-dc2a6703172e)

1. Does not work if the language is set to `System default`
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/ea87794a-6259-43a9-b9f3-58939758032c)

1. Perhaps this could be because `System default` = `en-AU English (Australia)` which is not in the list?
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/8efe6cea-b3ae-4ba0-b1bb-3eeae45e925a)

**Comment by n-stein** [comment]
> 2\. Does not work if the language is set to `System default`

It works for me on system default, but that is because my default is en_US. Your default language is en_AU and we don't have an en_AU translation file with which to combine the common.po translations.

**Comment by ovari** [comment]
If the system default is undefined (unavailable etc), can MMEX default to en-US?

**Comment by n-stein** [comment]
> If the system default is undefined (unavailable etc), can MMEX default to en-US?

Makes sense, if we don't have a translation catalog for the system language the entire program will render in US English anyway.

**Comment by vomikan** [comment]
For some reason I don't see any changes.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/6836805/453791a1-9a2f-4485-920b-01abfec41ad8)

**Comment by vomikan** [comment]
One more issue

![image](https://github.com/moneymanagerex/moneymanagerex/assets/6836805/cb11df50-dfc7-4865-9104-94b73fa09686)
